File: 05/exercise2-objc/driver.m
#import "vehicle.h"
#import "taxi.h"
#import "truck.h"

int main(int argc, char* argv[]) {
	Vehicle *car = [[Vehicle alloc] AmountOfDoor: 2 AmountOfCylinder: 6 TransmissionType: 3 Color: @"blue" FuelLevel: 14.6];
	Taxi *taxi = [[Taxi alloc] AmountOfDoor: 4 AmountOfCylinder: 6 TransmissionType: 5 Color: @"yellow" FuelLevel: 3.3 PresenceOfCustomer: NO];
	Truck *truck = [[Truck alloc] AmountOfDoor: 2 AmountOfCylinder: 16 TransmissionType: 8 Color: @"black" FuelLevel: 7.54 IfLoaded: YES];
	[car printDetail];
	printf("\n");
	[taxi printDetail];
	printf("\n");
	[truck printDetail];
	return 0;
}

File: 05/exercise2/driver.cpp
#include "vehicle.h"
#include "taxi.h"
#include "truck.h"

using namespace std;

int main(int argc, char* argv[]) {
	Vehicle *car = new Vehicle(2, 6, 3, "blue", 14.6);
	Taxi *taxi = new Taxi(4, 6, 5, "yellow", 3.3, false);
	Truck *truck = new Truck(2, 16, 8, "black", 7.54, true);
	cout << "Regular Method Calling:" << endl;
	car->printDetail();
	cout << endl;
	taxi->printDetail();
	cout << endl;
	truck->printDetail();
	cout << endl;
	cout << "Operator Overloading:" << endl;
	cout << *car << endl;
	cout << *taxi << endl;
	cout << *truck << endl;

	return 0;
}